Content.Server/Access/Systems/IdCardConsoleSystem.cs

```diff
@@ -151,25 +151,32 @@ private void TryWriteToTargetId(EntityUid uid,
         if (oldTags.SequenceEqual(newAccessList))
             return;
 
-        // I hate that C# doesn't have an option for this and don't desire to write this out the hard way.
-        // var difference = newAccessList.Difference(oldTags);
-        var difference = newAccessList.Union(oldTags).Except(newAccessList.Intersect(oldTags)).ToHashSet();
+        var ableToModify = component.AccessLevels.ToHashSet();
+
+        var addedTags = newAccessList.Except(oldTags).Select(tag => tag).ToHashSet();
+        var removedTags = oldTags.Except(newAccessList).Select(tag => tag).ToHashSet();
+
+        //Probably there is a cleaner way to do this, visible difference is the difference between the new access and the old access the user of the computer could see
+        var difference = addedTags.Union(removedTags).ToHashSet();
+        var visibleDifference = difference.Intersect(ableToModify).ToHashSet();
+        var hiddenAccess = difference.Except(visibleDifference).ToList();
```

Comment:
It makes more sense to calculate this as `difference.Except(ableToModify)` instead.